sanic/app.py

```diff
@@ -247,6 +247,12 @@ def websocket(self, uri, host=None, strict_slashes=None,
         def response(handler):
             async def websocket_handler(request, *args, **kwargs):
                 request.app = self
+                if not getattr(handler, '__blueprintname__', False):
```

Comment:
will you please make this (and used in converted_response_type) portion a function.
Also I would drop 'else' statement there, so it looks like:
def _get_endpoint(handler):
      if not getattr(handler, '__blueprintname__', False):
         return handler.__name__
      return getattr(handler,
                                       '__blueprintname__',
                                      '') + handler.__name__

and used as 
request.endpoint = _get_endpoint(handler)